src/validate.rs

```diff
@@ -778,16 +778,40 @@ fn validate_repos(data: &Data, errors: &mut Vec<String>) {
 
 /// Validate that branch protections make sense in combination with used bots.
 fn validate_branch_protections(data: &Data, errors: &mut Vec<String>) {
+    let github_teams = data.github_teams();
+
     wrapper(data.repos(), errors, |repo, _| {
         let bors_used = repo.bots.iter().any(|b| matches!(b, Bot::Bors));
         for protection in &repo.branch_protections {
-            if bors_used && protection.required_approvals.is_some() {
-                bail!(
-                    r#"repo '{}' uses bors and its branch protection for {} uses the `required-approvals` attribute;
+            if bors_used {
+                if protection.required_approvals.is_some() {
+                    bail!(
+                        r#"repo '{}' uses bors and its branch protection for {} uses the `required-approvals` attribute;
 please remove the attribute when using bors"#,
-                    repo.name,
-                    protection.pattern,
-                );
+                        repo.name,
+                        protection.pattern,
+                    );
+                }
+                if !protection.allowed_merge_teams.is_empty() {
+                    bail!(
+                        r#"repo '{}' uses bors and its branch protection for {} uses the `allowed-merge-teams` attribute;
+please remove the attribute when using bors"#,
+                        repo.name,
+                        protection.pattern,
+                    );
+                }
+            }
```

Comment:
Note: it was hard to tell that the check below this was not in the `if bors_used` block. Not sure if moving things out to separate functions would actually make things better so it might just be something we need to stick with, but I thought I'd mention it.

Reply:
I moved the check above the bors check, that should make it clearer.